pkg/storage/repo/storage.go

```diff
@@ -0,0 +1,269 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+	"io"
+	log1 "log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"net/http"
+	"net/url"
+	"os/exec"
+	"strings"
+
+	// git "github.com/gogs/git-module"
+	"github.com/rs/zerolog/log"
+
+	"github.com/bacalhau-project/bacalhau/pkg/clone"
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/publisher/estuary"
+	"github.com/bacalhau-project/bacalhau/pkg/storage"
+	apicopy "github.com/bacalhau-project/bacalhau/pkg/storage/ipfs"
+	"github.com/bacalhau-project/bacalhau/pkg/system"
+)
+
+type Response struct {
+	CID string
+}
+
+type StorageProvider struct {
+	LocalDir      string
+	EstuaryAPIKey string
+	CloneClient   *clone.Clone
+	IPFSClient    *apicopy.StorageProvider
+}
+
+func NewStorage(cm *system.CleanupManager, IPFSapiclient *apicopy.StorageProvider, EstuaryAPIKey string) (*StorageProvider, error) {
+	c, err := clone.NewCloneClient()
+	if err != nil {
+		return nil, err
+	}
+	dir, err := os.MkdirTemp(config.GetStoragePath(), "bacalhau-repo")
+	if err != nil {
+		return nil, err
+	}
+	cm.RegisterCallback(func() error {
+		if err := os.RemoveAll(dir); err != nil {
+			return fmt.Errorf("unable to remove storage folder: %w", err)
+		}
+		return nil
+	})
+	storageHandler := &StorageProvider{
+		LocalDir:      dir,
+		EstuaryAPIKey: EstuaryAPIKey,
+		IPFSClient:    IPFSapiclient,
+		CloneClient:   c,
+	}
+	log.Debug().Msgf("Repo download driver created with output dir: %s", dir)
+	return storageHandler, nil
+}
+
+func (sp *StorageProvider) IsInstalled(context.Context) (bool, error) {
+	err := checkGitLFS()
+	return err == nil, err
+}
+
+func (sp *StorageProvider) HasStorageLocally(context.Context, model.StorageSpec) (bool, error) {
+	return false, nil
+}
+
+// Could do a HEAD request and check Content-Length, but in some cases that's not guaranteed to be the real end file size
+func (sp *StorageProvider) GetVolumeSize(context.Context, model.StorageSpec) (uint64, error) {
+	return 0, nil
+}
+
+//nolint:gocyclo
+func (sp *StorageProvider) PrepareStorage(ctx context.Context, storageSpec model.StorageSpec) (storage.StorageVolume, error) {
+	_, span := system.GetTracer().Start(ctx, "pkg/storage/repo/repo.PrepareStorage")
+	defer span.End()
+	repoURL := storageSpec.Repo
+	var err error
+	_, err = clone.IsValidGitRepoURL(repoURL)
+	// fmt.Printf("%+v", storageSpec)
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+	// # create a tmp directory
+	outputPath, err := os.MkdirTemp(sp.LocalDir, "*")
+	fmt.Printf("Output Path: %s\n", outputPath)
```

Comment:
Use `log.Debug().Str("Path", outputPath).Send()`, don't print directly to stdout.